src/ProjectTempoListener.cpp

```diff
@@ -0,0 +1,42 @@
+#include "ProjectTempoListener.h"
+#include "Project.h"
+#include "ProjectTimeSignature.h"
+#include "WaveTrack.h"
+
+#include <chrono>
+#include <fstream>
+
+static const AttachedProjectObjects::RegisteredFactory key {
+   [](AudacityProject& project) {
+      return std::make_shared<ProjectTempoListener>(
+         project, TrackList::Get(project));
+   }
+};
+
+ProjectTempoListener::ProjectTempoListener(
+   AudacityProject& project, TrackList& trackList)
+    : mTrackList { trackList }
+    , mTrackListSubstription { trackList.Subscribe(
+         [&project](const TrackListEvent& event) {
+            if (event.mType == TrackListEvent::ADDITION)
+            {
+               const auto tempo = ProjectTimeSignature::Get(project).GetTempo();
+               if (const auto track = event.mpTrack.lock())
+                  track->OnProjectTempoChange(tempo);
+            }
+         }) }
+{
+   auto& timeSignature = ProjectTimeSignature::Get(project);
+   OnProjectTempoChange(timeSignature.GetTempo());
```

Comment:
I think this step of construction is not needed.  There will not yet be any tracks.